ggml/src/ggml-opencl2/ggml-opencl2.cpp

```diff
@@ -0,0 +1,3923 @@
+        GGML_ASSERT(ggml_is_contiguous(src0));
+
+        // src1 is a row
+        GGML_ASSERT(ne11 == 1);
+
+        bcast_row = true;
+        nb = ne00 / 4;
+        kernel = backend_ctx->kernel_add_row;
+
+        CL_CHECK(clSetKernelArg(kernel, 0, sizeof(cl_mem), &extra0->data_device));
+        CL_CHECK(clSetKernelArg(kernel, 1, sizeof(int),    &offset0));
+        CL_CHECK(clSetKernelArg(kernel, 2, sizeof(cl_mem), &extra1->data_device));
+        CL_CHECK(clSetKernelArg(kernel, 3, sizeof(int),    &offset1));
+        CL_CHECK(clSetKernelArg(kernel, 4, sizeof(cl_mem), &extrad->data_device));
+        CL_CHECK(clSetKernelArg(kernel, 5, sizeof(int),    &offsetd));
+        CL_CHECK(clSetKernelArg(kernel, 6, sizeof(int),    &nb));
+    } else {
+        kernel = backend_ctx->kernel_add;
+
+        CL_CHECK(clSetKernelArg(kernel, 0, sizeof(cl_mem), &extra0->data_device));
+        CL_CHECK(clSetKernelArg(kernel, 1, sizeof(int),    &offset0));
+        CL_CHECK(clSetKernelArg(kernel, 2, sizeof(cl_mem), &extra1->data_device));
+        CL_CHECK(clSetKernelArg(kernel, 3, sizeof(int),    &offset1));
+        CL_CHECK(clSetKernelArg(kernel, 4, sizeof(cl_mem), &extrad->data_device));
+        CL_CHECK(clSetKernelArg(kernel, 5, sizeof(int),    &offsetd));
+        CL_CHECK(clSetKernelArg(kernel,  6, sizeof(int),   &ne00));
+        CL_CHECK(clSetKernelArg(kernel,  7, sizeof(int),   &ne01));
+        CL_CHECK(clSetKernelArg(kernel,  8, sizeof(int),   &ne02));
+        CL_CHECK(clSetKernelArg(kernel,  9, sizeof(int),   &ne03));
+        CL_CHECK(clSetKernelArg(kernel, 10, sizeof(int),   &nb00));
+        CL_CHECK(clSetKernelArg(kernel, 11, sizeof(int),   &nb01));
+        CL_CHECK(clSetKernelArg(kernel, 12, sizeof(int),   &nb02));
+        CL_CHECK(clSetKernelArg(kernel, 13, sizeof(int),   &nb03));
+        CL_CHECK(clSetKernelArg(kernel, 14, sizeof(int),   &ne10));
+        CL_CHECK(clSetKernelArg(kernel, 15, sizeof(int),   &ne11));
+        CL_CHECK(clSetKernelArg(kernel, 16, sizeof(int),   &ne12));
+        CL_CHECK(clSetKernelArg(kernel, 17, sizeof(int),   &ne13));
+        CL_CHECK(clSetKernelArg(kernel, 18, sizeof(int),   &nb10));
+        CL_CHECK(clSetKernelArg(kernel, 19, sizeof(int),   &nb11));
+        CL_CHECK(clSetKernelArg(kernel, 20, sizeof(int),   &nb12));
+        CL_CHECK(clSetKernelArg(kernel, 21, sizeof(int),   &nb13));
+        CL_CHECK(clSetKernelArg(kernel, 22, sizeof(int),   &ne0));
+        CL_CHECK(clSetKernelArg(kernel, 23, sizeof(int),   &ne1));
+        CL_CHECK(clSetKernelArg(kernel, 24, sizeof(int),   &ne2));
+        CL_CHECK(clSetKernelArg(kernel, 25, sizeof(int),   &ne3));
+        CL_CHECK(clSetKernelArg(kernel, 26, sizeof(int),   &nb0));
+        CL_CHECK(clSetKernelArg(kernel, 27, sizeof(int),   &nb1));
+        CL_CHECK(clSetKernelArg(kernel, 28, sizeof(int),   &nb2));
+        CL_CHECK(clSetKernelArg(kernel, 29, sizeof(int),   &nb3));
```

Comment:
Note that passing the strides as `int` instead of `uint64_t` is not a good idea. I guess for smaller models it will be fine, but it will inevitably come back to bite you one day. Might want to consider a change similar to https://github.com/ggerganov/llama.cpp/pull/10238

Reply:
Ah. Yeah. Thanks for point this. Replacing ...